### PR TITLE
DeepSSM 

### PR DESCRIPTION
**Pull request by jadie1**
This pull request:
- adds DataAugmentationUtils to Python packages (issue #581 and issue #625)
- adds DeepSSMUtils to Python packages (issue #581)
- updates conda_installs.sh to include DataAugmentationUtils, DeepSSMUtils, Pytorch (GPU if available otherwise CPU version), and bokeh (issue #481)
- adds a new use case: deep_ssm.py (issue #551)
- adds a new mesh function: SurfaceToSurfaceDistance (issue #527)
- adds documentaton (issue #552)
	- docs/deep-learning/deep-ssm.md
	- docs/deep-learning/data-augmentaion.md
	- docs/use-cases/deep-ssm-femur.md

### CODE REVIEW

**Comment by jadie1** [comment]
To test I recommend:
1. Run conda_installs.sh
2. Run build_dependencies.sh and cmake again if you haven't already for the VTK 9.0 upgrade
3. Run the use case tiny test:
```
cd ShapeWorks/Examples/Python/
python RunUseCase.py --use_case deep_ssm --tiny_test
```
If you get the Pytorch running on CPU error, then conda_installs.sh picked the wrong CUDA version. Install the correct one with:
`conda install -n shapeworks pytorch torchvision cudatoolkit=CORRECT_VERSION_NUM -c pytorch`
4. Check over the documentation (@riddhishb I could use your advice here)

You can also run the full use case, it takes about a day to run. The tiny_test does not skip any steps but the results are bad so to see the accuracy of the results the full use case needs to be run. 

I tested the surface to surface distance tool pretty thoroughly, here is how to use it:
`SurfaceToSurfaceDistance --inputA path_to_input_mesh_A --inputB path_to_input_mesh_B --outputA path_to_output_mesh_A --outputB path_to_output_mesh_B`

The input meshes need to be VTK. The output meshes are the same as the input but they have a scalar field on them with the distance value, this can be visualized as a heatmap in ParaView. This is just the base functionality, SurfaceToSurfaceDistance will be updated next release to be more flexible and do more (issue #708)

File: Python/DataAugmentationUtilsPackage/DataAugmentationUtils/Sampler.py
Comment (akenmorris) [inline Python/DataAugmentationUtilsPackage/DataAugmentationUtils/Sampler.py:65]: Typo, Fittting -> Fitting.

File: docs/deep-learning/deep-ssm.md
Comment (riddhishb) [inline docs/deep-learning/deep-ssm.md:2]: DeepSSM is the framework which includes data-augmentation and then the prediciton is made by a CNN model trained on image,correspondence pairs.

File: docs/deep-learning/deep-ssm.md
Comment (riddhishb) [inline docs/deep-learning/deep-ssm.md:5]: otherwise => other Particle distribution models

File: docs/deep-learning/deep-ssm.md
Comment (riddhishb) [inline docs/deep-learning/deep-ssm.md:20]: PBM ? or PDM ?

File: docs/deep-learning/deep-ssm.md
Comment (riddhishb) [inline docs/deep-learning/deep-ssm.md:38]: command format

File: docs/deep-learning/deep-ssm.md
Comment (riddhishb) [inline docs/deep-learning/deep-ssm.md:5]: PBM/PDM ? 
Also the first time the abbreviation is declared give the full form, after that you can use it freely/

File: docs/use-cases/deep-ssm-femur.md
Comment (riddhishb) [inline docs/use-cases/deep-ssm-femur.md:1]: I think this use case should be in use-cases/deep-learning, right ?

**Review by riddhishb — CHANGES_REQUESTED** [review]
Hi @jadie1 I have reviewed the documentation, the comments are in the diff.

**Comment by akenmorris** [comment]
This is the output I get on the tinytest on mac, is this right?

Screen Shot 2020-09-28 at 2 31 58 PM

**Comment by jadie1** [comment]
@akenmorris Yeah that's right! This is just a preliminary visualization, we have an issue to improve it for the next release.

File: docs/use-cases/deep-ssm-femur.md
Comment (jadie1) [reply docs/use-cases/deep-ssm-femur.md:1]: @sheryjoe What do you think? Should we have a deep-learning folder inside the use-cases folder?

**Comment by jadie1** [comment]
@sheryjoe I have addressed most of Riddhishs comments on the documentation now if you'd like to look over it.

**Comment by riddhishb** [comment]
I get this, different from Alan's plot
Screenshot_2020-09-28_19-54-58

**Comment by jadie1** [comment]
Looks good. The real examples (blue) are the same and the augmented (red) are sampled randomly so we expect them to be different.

**Comment by iyerkrithika21** [comment]
image
Output on windows.

**Comment by akenmorris** [comment]
Ok, the windows build is fixed.

**Comment by akenmorris** [comment]
Are @riddhishb ’s requested changes resolved?

**Comment by jadie1** [comment]
@akenmorris working on them now, then it will be ready to merge.

File: docs/deep-learning/data-augmentation.md
Comment (jadie1) [reply docs/deep-learning/data-augmentation.md:37]: It is both, we find the number of components that minimize AIC and the number of components that minimize BIC and average them.

**Comment by jadie1** [comment]
@sheryjoe @akenmorris I finished making the updates Riddhish suggested for the documentation and I tested everything including the surface to surface distance tool with VTK 8.2 and it is all working (thank you Alan!) I think this is ready to merge!

**Comment by akenmorris** [comment]
Great. I will just wait for the ok from the checks and then merge.

**Pull request merged**
